apps/nextjs/src/components/StoreComponents/ModerationSync.tsx

```diff
@@ -0,0 +1,41 @@
+import { useEffect } from "react";
+
+import { useModerationStore } from "@/stores/moderationStore";
+import { trpc } from "@/utils/trpc";
+
+export const ModerationSync = ({ id }: { id: string }) => {
+  const {
+    data: moderations,
+    isLoading: isModerationsLoading,
+    refetch: refetchModerations,
+  } = trpc.chat.appSessions.getModerations.useQuery(
+    { id },
+    {
+      refetchOnMount: true,
+      refetchOnWindowFocus: true,
+      staleTime: 0,
+    },
+  );
+
+  const setModeration = useModerationStore(
+    (state) => state.updateModerationState,
+  );
+  const setIsModerationsLoading = useModerationStore(
+    (state) => state.setIsModerationsLoading,
+  );
+
+  useEffect(() => {
+    setIsModerationsLoading(isModerationsLoading);
+  }, [isModerationsLoading, setIsModerationsLoading]);
+
+  useEffect(() => {
+    if (moderations !== undefined) {
+      setModeration(moderations || []);
+    }
+  }, [moderations, setModeration]);
+
+  useEffect(() => {
+    void refetchModerations();
+  }, [refetchModerations]);
```

Comment:
I think this is fine to get things working, but we will be trying to move away from having logic and data loading powered by hooks and effects. I think all of this behaviour should live in an action function, though that's something we can look at in a follow up